### PR TITLE
Add github action to run test in template

### PR DESCRIPTION
**Pull request by tianfeng92**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -51,3 +51,29 @@ jobs:
...
         run: bundle/node_modules/.bin/npm -v
+
+  test_template:
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v2
+
+      - uses: actions/setup-go@v2
```
Comment (alexplischke) [inline .github/workflows/test.yml:62]: Why is go necessary? 🤔

Comment (tianfeng92) [reply .github/workflows/test.yml:62]: you're right 🤦

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -51,3 +51,36 @@ jobs:
...
+          skip-run: true
+
+      # There is discrepencies between uid used to clone and uid in container,
+      # As a temporary measure, we chmod files to allow test outputs to be written.
+      - name: Workaround for container permissions
+        run: sudo chmod -R 777 .saucetpl/
+
+      # Run test in template
+      - run: |
+          docker build -t saucelabs/stt-cypress-mocha-node:local . > /dev/null 2>&1
```
Comment (FriggaHel) [inline .github/workflows/test.yml:77]: You can also consider building it with the action `docker/build-push-action@v1` like in `build` job.
(No changes required)

**Review by FriggaHel — APPROVED** [review]

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -51,3 +51,37 @@ jobs:
...
+          cd .saucetpl
+          yq e -i '.docker.image = "saucelabs/stt-cypress-mocha-node:local"' .sauce/config.yml
+          yq e -i '.docker.fileTransfer = "copy"' .sauce/config.yml
+          yq e -i '.cypress.version = "latest"' .sauce/config.yml
```
Comment (FriggaHel) [inline .github/workflows/test.yml:81]: I think this may not be required since image is provided.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -51,3 +51,37 @@ jobs:
...
+      # As a temporary measure, we chmod files to allow test outputs to be written.
+      - name: Workaround for container permissions
+        run: sudo chmod -R 777 .saucetpl/
+
+      # Run test in template
+      - run: |
+          docker build -t saucelabs/stt-cypress-mocha-node:local . > /dev/null 2>&1
+          cd .saucetpl
+          yq e -i '.docker.image = "saucelabs/stt-cypress-mocha-node:local"' .sauce/config.yml
+          yq e -i '.docker.fileTransfer = "copy"' .sauce/config.yml
```
Comment (FriggaHel) [inline .github/workflows/test.yml:80]: Why is this required ? `fileTransfer: mount` is not working on github actions ?

File: .github/workflows/test.yml (hunk 3 above)
Comment (tianfeng92) [reply .github/workflows/test.yml:81]: you're right

Comment (tianfeng92) [reply .github/workflows/test.yml]: oh it's still required a version since it's
```
version: ##VERSION##
```
